File: Data Structures & Algorithms/max-consecutive-ones/submission-1.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        ans = 0
        res = 0
        for i, n in enumerate(nums):
            if n == 0:
                res = max(res, ans)
                ans = 0
            else:
                ans += 1
        return max(res, ans)

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int k = 1, k2 = 1, c = 0;
        for(int i = 0; i < nums.size(); i++) {
            k *= nums[i];
            if (nums[i] == 0) c++;
        }
        if (c > 1) return vector<int>(nums.size(), 0);
        vector<int> ans;
        for(int i = 0; i < nums.size(); i++) {
            if (nums[i] != 0) ans.push_back(k / nums[i]);
            else {
                for(int i = 0; i < nums.size(); i++) {
                    if (nums[i] != 0) k2 *= nums[i];
                }
                ans.push_back(k2);
            }
        }
        return ans;
    }
};
